.github/workflows/python-app.yml

```diff
@@ -21,13 +24,35 @@ jobs:
 
     - name: Install dependencies and build project
       run: |
-        pip install prompt-toolkit
-        pip install jsonschema
-        pip install pytest
-        pip install pytest-cov
-        pip install -e .
-
-    - name: Run tests
-      run: pytest --cov=src/Models
+          pip install prompt-toolkit
```

Comment:
-e . installs prompt-toolkit, jsonschema, and pytest. You shouldn't need to pip install them as well, just run -e . before installing pytest-cov because I am assuming it has pytest as a dependency.

Reply:
if it doesn't install pytest, we can change setup.py to fix that. I think it's in an optional dependency group right now, based on [this](https://stackoverflow.com/questions/75490231/pip-pyproject-toml-can-optional-dependency-groups-require-other-optional-depend) so post, we should be able to include optional dependencies with the command ``pip install -e .[dependency name]``, so leaving it as optional should be fine.

Reply:
i think the dependency name would be ``tool.poetry.dev_dependencies``, though I'm not certain.

Reply:
Removed redundant package calls, kept pytest as it is not installed by default. Couldn't get tool.poetry.dev_dependencies working but the manual install call works fine for now.